src/dev/flang/fe/SourceModule.java

```diff
@@ -1110,7 +1119,10 @@ public FeatureAndOuter lookupType(SourcePosition pos,
         var curOuter = outer;
         var type_fs = new List<AbstractFeature>();
         var nontype_fs = new List<AbstractFeature>();
-        var fs = lookup(outer, name, null, traverseOuter);
+        var fs = lookup(outer, name, null, traverseOuter)
+                   .stream()
+                   .filter(fo -> typeVisible(pos._sourceFile, fo._feature))
+                   .collect(Collectors.toList());
         for (var fo : fs)
           {
             var f = fo._feature;
```

Comment:
Same here, maybe add an `if (typeVisible(...))` in the loop instead of using the stream.